scripts/xtensa-build-zephyr.py

```diff
@@ -542,7 +542,10 @@ def build_platforms():
 					"or is not a directory")
 
 			# set variables expected by zephyr/cmake/toolchain/xcc/generic.cmake
-			os.environ["ZEPHYR_TOOLCHAIN_VARIANT"] = "xcc"
+			if not os.getenv("ZEPHYR_TOOLCHAIN_VARIANT"):
+				raise RuntimeError(f"Platform {platform} uses Xtensa toolchain."
+					"\nPlease set ZEPHYR_TOOLCHAIN_VARIANT environmental variable to"
+					"'xcc' or 'xcc-clang'")
```

Comment:
Can you please provide a default value here?
```suggestion
if not os.getenv("ZEPHYR_TOOLCHAIN_VARIANT"):
    os.environ["ZEPHYR_TOOLCHAIN_VARIANT"] = "xcc-clang"
```

Reply:
I don't think default value is a good idea here especially one set to xt-clang. Script end-user if not set this variable could be trying to compile TGL with xt-clang what is possible but what we do not support (well at least I had not checked it for compiler errors/warnings). Also TGL does not have xcc-clang toolchain enabled in Zephyr board yaml file. I'm not sure what would happen.

Reply:
> Is this value needed by some CI systems?

If it's optional then why fail when it's missing?

> if not set this variable could be trying to compile TGL with xt-clang what is possible but what we do not support [...]

It's OK to fail in unsupported cases.  It's not OK to require a new, "evil" environment variable for the default case.

>  I'm not sure what would happen.

If the error message for some unsupported case is too cryptic, the few people who want/need that case can always submit a better error message.

> Also TGL does not have xcc-clang toolchain enabled in Zephyr board yaml file.

BTW the `platform_list` has always provided platform-specific default values. In fact `platform_list` is probably the main value of this script: platform-specific default values that allow building multiple platforms at once without any painful configuration tricks. It will be a real regression to lose that.

Reply:
Hmm, yeah you are right.
Maybe approach with expanding platform_list values would be much better solution (we just did not have to distinguish between Xtensa compilers before). I'll just add new field to platform_list that would provide info that for TGL default toolchain is xt-xcc and for MTL is xt-clang. That would make much more sense.

Reply:
@marc-hb does this looks better now?